### ISSUE TITLE
[FEA] Provide option to read fields with mixed types

### ISSUE THREAD

**Issue by GregoryKimball** [rapidsai/cudf#11947]
**Is your feature request related to a problem? Please describe.**
The `cudf_experimental` JSON reader does not faithfully read fields with mixed nesting. 

If the field contains `scalar` and `list` then the `scalar` values are set to null:
```
>>> json_str = '[{"a":1},{"a":[1]}]'
>>> cudf.read_json(json_str, engine='cudf')
      a
0  None
1   [1]
```

If the field contains `scalar` and `struct` then the `scalar` values are set to null:
```
json_str = '[{"a":1},{"a":{}}]'
>>> cudf.read_json(json_str, engine='cudf')
      a
0  None
1    {}
```

If the field contains different levels of nested lists, the shallower lists are set to null:
```
>>> json_str = '[{"a":[1]},{"a":[[1]]}]'
>>> cudf.read_json(json_str, engine='cudf')
        a
0  [None]
1   [[1]]
```

If the field contains `list` and `struct` types, we throw:
```
>>> json_str = '[{"a":[1]},{"a":{"b":1}}]'
>>> cudf.read_json(json_str, engine='cudf')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/conda/lib/python3.10/site-packages/cudf/io/json.py", line 111, in read_json
    df = libjson.read_json(
  File "json.pyx", line 50, in cudf._lib.json.read_json
  File "json.pyx", line 138, in cudf._lib.json.read_json
RuntimeError: CUDF failure at: /opt/conda/conda-bld/work/cpp/src/io/json/json_column.cu:576: A mix of lists and structs within the same column is not supported
```

**Describe the solution you'd like**
We've talked about a few solutions:
* model the column as a struct that represents a union column
* add a union type to libcudf
* read the field as multiple columns and put a subscript in the name
* coerce the field's values to strings instead of trying to represent the mixed nesting

**Describe alternatives you've considered**
Without a workaround, JSON inputs including a field with mixed types will not be readable.

**Additional context**
This item still needs some design work. Please comment and share a data sample if it is impacting you.

**Comment by isVoid** [comment]
It is common in GeoJson to have different level of nested list data in the same column. Here's a column of a polygon and a linestring. (You can visualize this in https://geojson.io/)
[map.zip](https://github.com/rapidsai/cudf/files/10249932/map.zip)

**Comment by GregoryKimball** [comment]
After some internal discussion we are favoring the option:
> model the column as a struct that represents a union column

Here is an example of what this representation could look like:
```
[
  { "a": "foo" },
  { "a": { "b": "bar" } }
]
```

```
>>> df
                                             a
0         {'0': 'foo', '1': None, 'offset': 0}
1  {'0': None, '1': {'b': 'bar'}, 'offset': 1}
>>> df['a'].dtype
StructDtype({'0': dtype('O'), '1': StructDtype({'b': dtype('O')}), 'offset': dtype('int64')})
```

**Comment by pdufour** [comment]
This would be great :) Also running into this with this file:

(value gets set as None for first two times, then keeps the empty object)
```
{"Type":"insert","Key":[1],"SeqNo":1,"Timestamp":1,"Fields":[{"Name":"added_id","Value":1},{"Name":"row_key","Value":"//1=="},{"Name":"column_key","Value":"2"},{"Name":"ref_key","Value":1},{"Name":"updated","Value":"2020-12-08T02:15:07.664543Z"},{"Name":"body","Value":{}}]}
```

**Comment by GregoryKimball** [comment]
Also see #14239

**Comment by GregoryKimball** [comment]
Closed by #14572
